test/ClientSpec.hs

```diff
@@ -32,6 +49,7 @@ spec =
         someHeaderLength (Just "xyz987") `shouldRespondWith_` 200
         _400s `shouldRespondWith_` 400
         _500s `shouldRespondWith_` 500
+        dontSucceed _500s
```

Comment:
I think you forgot to use `decodingError`

```haskell
dontSucceed decodingError
```

The change to `dontSucceed` I mentioned above is needed for this test to work though

Reply:
Here I just wanted to test `dontSucceed` itself. I don't think we can use `dontSucceed` to check decodingError because the server indeed sends a 200, but decoding fails post response.